src/js/components/ClusterDropdown.js

```diff
@@ -0,0 +1,181 @@
+import * as React from "react";
+import mixin from "reactjs-mixin";
+import { Dropdown } from "reactjs-components";
+import { StoreMixin } from "mesosphere-shared-reactjs";
+import { MountService } from "foundation-ui";
+import { routerShape } from "react-router";
+
+import MesosSummaryStore from "#SRC/js/stores/MesosSummaryStore";
+import MetadataStore from "#SRC/js/stores/MetadataStore";
+import SidebarActions from "#SRC/js/events/SidebarActions";
+import ClipboardTrigger from "#SRC/js/components/ClipboardTrigger";
+
+function ClusterDropdownTrigger({ onTrigger, children }) {
+  return (
+    <span className="header-bar-dropdown-trigger" onClick={onTrigger}>
+      {children}
+    </span>
+  );
+}
+const METHODS_TO_BIND = ["handleItemSelect", "handleTextCopy"];
+export default class ClusterDropdown extends mixin(StoreMixin) {
```

Comment:
🎨 if you're refactoring anyway, why not getting rid of the StoreMixing? :trollface:

Reply:
Agree!